pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
OK. I think the deleted object should be stored in the cache indexed by its acutal RV (otherwise the cache is pointless), but the RV sent to the user ought to be the current index. make sense?